lib/matplotlib/tests/test_axes.py

```diff
@@ -4603,56 +4603,68 @@ def test_hist_stacked_bar():
-
-    for i, (xi, c) in enumerate(zip(x, colors)):
-        axr6.hist(xi, bins=bins, histtype="stepfilled", color=c, zorder=(len(x)-i)/2)
+    (axt1, axt2) = fig_test.subplots(2)
+    (axr1, axr2) = fig_ref.subplots(2)
+
+    for histtype, axt, axr in [("stepfilled", axt1, axr1), ("step", axt2, axr2)]:
+        _, bins, _ = axt.hist(x, bins=10, histtype=histtype, **kwargs)
+
+        kw, values = next(iter(kwargs.items()))
+        for i, (xi, value) in enumerate(zip(x, values)):
+            axr.hist(xi, bins=bins, histtype=histtype, **{kw: value},
+                     zorder=(len(x)-i)/2)
+
+
+@pytest.mark.parametrize('kwargs, patch_face, patch_edge',
+                         [({'histtype': 'stepfilled', 'color': 'r',
+                            'facecolor': 'y', 'edgecolor': 'g'}, 'y', 'g'),
+                          ({'histtype': 'step', 'color': 'r',
+                            'facecolor': 'y', 'edgecolor': 'g'}, 'none', 'g'),
+                          ({'histtype': 'stepfilled', 'color': 'r',
+                            'edgecolor': 'g'}, 'r', 'g'),
+                          ({'histtype': 'step', 'color': 'r',
+                            'edgecolor': 'g'}, 'none', 'g'),
+                          ({'histtype': 'stepfilled', 'color': 'r',
+                            'facecolor': 'y'}, 'y', 'k'),
+                          ({'histtype': 'step', 'color': 'r',
+                            'facecolor': 'y'}, 'none', 'r'),
+                          ({'histtype': 'stepfilled',
+                            'facecolor': 'y', 'edgecolor': 'g'}, 'y', 'g'),
+                          ({'histtype': 'step', 'facecolor': 'y',
+                            'edgecolor': 'g'}, 'none', 'g'),
+                          ({'histtype': 'stepfilled', 'color': 'r'}, 'r', 'k'),
+                          ({'histtype': 'step', 'color': 'r'}, 'none', 'r'),
+                          ({'histtype': 'stepfilled', 'facecolor': 'y'}, 'y', 'k'),
+                          ({'histtype': 'step', 'facecolor': 'y'}, 'none', 'b'),
+                          ({'histtype': 'stepfilled', 'edgecolor': 'g'}, 'b', 'g'),
+                          ({'histtype': 'step', 'edgecolor': 'g'}, 'none', 'g'),
+                          ({'histtype': 'stepfilled'}, 'b', 'k'),
+                          ({'histtype': 'step'}, 'none', 'b')])
+def test_hist_color_semantics(kwargs, patch_face, patch_edge):
+    _, _, patches = plt.figure().subplots().hist([1, 2, 3], **kwargs)
+    # When the expected edgecolor is blue, it corresponds to the
+    # first color of the default color cycle
+    # When the expected edgecolor is black, it corresponds to the
+    # patch.edgecolor rcParam
+    # When the expected facecolor is blue, it corresponds to the
+    # patch.facecolor rcParam or the first color of the color cycle
+    assert all(mcolors.same_color(p.get_facecolor(), patch_face)
+               if patch_face != 'none' else p.get_facecolor()[3] == 0
+               for p in patches)
```

Comment:
I will have to special case *none* as the facecolor/edgecolor that I get from the patch is never (0,0,0,0) and is some rgb value with a 0 alpha. For example in the third case on the table:
```kwargs = {"histtype": "step", "color": "r", "facecolor": "y", "edgecolor": "g"}```
```[mcolors.to_rgba(p.get_facecolor()) for p in patches]``` gives ```[(0.75, 0.75, 0.0, 0.0)]```

Reply:
> I'm confused b/c you're getting passed a facecolor and edgecolor so I don't understand why anything should be none

The behavior for 'step' is to not use a facecolor even when it is mentioned, as there is no *fill* in this type of plot. Hence the *none*

Reply:
> Also this has me wondering what's going on when we set "none" and if that's related to #28475

I don't think so, as we are not dealing with empty arrays here, but I'd love to take that one up in the future

Reply:
> Well so that one isn't an empty array issue, it's a "how is "none" processed" issue.
> 
> Which sounds like here, whats going on in step is that it sets whatever was passed in as the facecolor to have 0 alpha. Basically, I wonder if in that case, patch_facecolor (so expected face_color) is (facecolor, 0).

OK, i thought of doing this but it seemed like it would complicate things, ill make this change

Reply:
At first glance, it seemes like 'none' would be a good indicator of *the color is transparent*, but using this way is actually not that long and makes the test a lot more readable